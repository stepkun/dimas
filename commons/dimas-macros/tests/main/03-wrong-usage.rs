// Copyright Â© 2024 Stephan Kunz

//! Test wrong usage of main macro

#[dimas_macros::main]
pub struct Test {}

#[dimas_macros::main]
fn not_main() {}

#[dimas_macros::main]
fn main() {}