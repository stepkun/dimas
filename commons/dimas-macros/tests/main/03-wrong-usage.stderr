error: expected `fn`
 --> tests/main/03-wrong-usage.rs:6:5
  |
6 | pub struct Test {}
  |     ^^^^^^

error: macro can only be used for main function
 --> tests/main/03-wrong-usage.rs:9:4
  |
9 | fn not_main() {}
  |    ^^^^^^^^

error: missing `async` keyword in function declaration
  --> tests/main/03-wrong-usage.rs:12:1
   |
12 | fn main() {}
   | ^^
